### PR TITLE
[WFCORE-6658] Convert ControlledProcessStateService to the current MS…

### PR DESCRIPTION
**Pull request by bstansberry**
…C API

https://issues.redhat.com/browse/WFCORE-6658

### CODE REVIEW

**Comment by bstansberry** [comment]
@ropalka FYI.

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Server.java [hunk 1]
```diff
@@ -221,6 +230,7 @@ private synchronized void establishModelControllerClient(ControlledProcessState.
         if (state != ControlledProcessState.State.STOPPING && state != ControlledProcessState.State.STOPPED && serviceContainer != null) {
             ModelControllerClientFactory clientFactory;
             try {
+                // TODO replace this in start() with the ServiceActivator approach we used to capture the ProcessStateNotifier
```
Comment (yersan) [inline bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Server.java:233]: I've just added WFCORE-6662 as a follow up to cover this.

**Review by yersan — APPROVED** [review]
LGTM, @ropalka could you also take a quick look?

My very minor detail is that now the new SimpleService that provides the ProcessStateNotifier is not invoking `consumer.accept(null)` on the stop method as we did in ControlledProcessStateService:61, I don't think it could be an issue, but I am not certainly sure if that could cause any unexpected side effect. It is done when the service is being stopped, so it looks like safe.

**Comment by yersan** [comment]
@ropalka FYI in case you have missed it.

**Review by ropalka — APPROVED** [review]
LGTM

**Comment by yersan** [comment]
Thanks @bstansberry and @ropalka

**Pull request merged**